libraries/AP_HAL_Linux/Util_RPI.cpp

```diff
@@ -47,7 +48,16 @@ int UtilRPI::_check_rpi_version()
                 _rpi_version = 2;
             } else if (_rpi_version == 0) {
                 // RPi 1 doesn't have a number there, so sscanf() won't have read anything.
-                _rpi_version = 1;
+                ret = sscanf (buffer +12, "%s", &version[0]);
```

Comment:
what's the content of `/sys/firmware/devicetree/base/model` for rpi0 and rpi1? So.. you are adding this, but afaics rpi0 was already supported by just treating it as rpi1. Am I missing anything?

Reply:
yes that is the idea ... it is rpi0 yes it can be detectedby just treating it as rpi1 but it is not rpi1 and the code -in my opinion- needs to distinguish between them to keep it informative and reliable.